packages/currency-validator/src/currency-validator.ts

```diff
@@ -0,0 +1,37 @@
+export interface FieldValidatorInterface {
+  keydown(e: KeyboardEvent): void;
+}
+
+export class CurrencyValidator implements FieldValidatorInterface {
+  keydown(e: KeyboardEvent): void {
+    const char = e.key;
+
+    // if user is holding down command/ctrl for like select all, allow it
+    if (e.metaKey) {
+      return;
+    }
+
+    switch (char) {
+      case 'Tab':
+      case 'Delete':
+      case 'Backspace':
+      case 'ArrowLeft':
+      case 'ArrowRight':
+      case 'ArrowUp':
+      case 'ArrowDown':
+        return;
+    }
+
+    const input = e.target as HTMLInputElement;
+    const value = input.value;
+    const prefix = value.slice(0, input.selectionStart ?? 0);
+    const suffix = value.slice(input.selectionEnd ?? 0);
+    const newValue = `${prefix}${char}${suffix}`;
+    const regex = /^[0-9]+(\.[0-9]{0,2})?$/g; // currency regex xxxxx.xx
```

Comment:
This protects against values input one digit at a time, but does not protect against invalid values pasted in to the field. 
<img width="333" alt="Screen Shot 2020-09-11 at 8 31 35 AM" src="https://user-images.githubusercontent.com/39553/92925892-3936d800-f409-11ea-8898-596ab7a583ad.png">

Reply:
I will file a follow-up ticket for this. This is an existing issues and I'm just trying to extract this module from the main donation form.